Xamarin.Forms.Xaml.UnitTests/MSBuild/MSBuildTests.cs

```diff
@@ -0,0 +1,212 @@
+using System.IO;
+using System.Xml.Linq;
+using static Xamarin.Forms.Xaml.UnitTests.MSBuildXmlExtensions;
+
+namespace Xamarin.Forms.Xaml.UnitTests
+{
+	//This set of tests is for validating Xamarin.Forms.targets
+	[TestFixture]
+	public class MSBuildTests
+	{
+		const string XamarinFormsDefaultNamespace = "http://xamarin.com/schemas/2014/forms";
+		const string XamarinFormsXNamespace = "http://schemas.microsoft.com/winfx/2006/xaml";
+		static readonly string XamarinFormsTargets = Path.Combine ("..", "..", "..", "..", "..", ".nuspec", "Xamarin.Forms.targets");
+
+		static readonly string [] references = new []
+		{
+			"mscorlib",
+			"System",
+			"Xamarin.Forms.Core.dll",
+			"Xamarin.Forms.Xaml.dll",
+		};
+
+		string testDirectory;
+		string tempDirectory;
+		string intermediateDirectory;
+
+		[SetUp]
+		public void SetUp ()
+		{
+			testDirectory = TestContext.CurrentContext.TestDirectory;
+			tempDirectory = Path.Combine (testDirectory, "temp", TestContext.CurrentContext.Test.Name);
+			intermediateDirectory = Path.Combine (tempDirectory, "obj", "Debug");
+			Directory.CreateDirectory (tempDirectory);
+		}
+
+		[TearDown]
+		public void TearDown ()
+		{
+			if (Directory.Exists (tempDirectory))
+				Directory.Delete (tempDirectory, true);
+		}
+
+		XElement NewProject ()
+		{
+			var path = Path.GetTempFileName ();
+			var project = NewElement ("Project");
+
+			var propertyGroup = NewElement ("PropertyGroup");
+			propertyGroup.Add (NewElement ("Configuration").WithValue ("Debug"));
+			propertyGroup.Add (NewElement ("Platform").WithValue ("AnyCPU"));
+			propertyGroup.Add (NewElement ("OutputType").WithValue ("Library"));
+			propertyGroup.Add (NewElement ("OutputPath").WithValue ("bin\\Debug"));
+			propertyGroup.Add (NewElement ("TargetFrameworkVersion").WithValue ("v4.7"));
+			project.Add (propertyGroup);
+
+			var itemGroup = NewElement ("ItemGroup");
+			foreach (var assembly in references) {
+				var reference = NewElement ("Reference").WithAttribute ("Include", assembly);
+				if (assembly.EndsWith (".dll", StringComparison.OrdinalIgnoreCase))
+					reference.Add (NewElement ("HintPath").WithValue (Path.Combine ("..", "..", assembly)));
+				itemGroup.Add (reference);
+			}
+			project.Add (itemGroup);
+
+			project.Add (NewElement ("Import").WithAttribute ("Project", @"$(MSBuildBinPath)\Microsoft.CSharp.targets"));
+			project.Add (NewElement ("Import").WithAttribute ("Project", XamarinFormsTargets));
+			return project;
+		}
+
+		XElement AddFile (string name, string buildAction, string contents)
+		{
+			File.WriteAllText (Path.Combine (tempDirectory, name), contents);
+			var itemGroup = NewElement ("ItemGroup");
+			itemGroup.Add (NewElement (buildAction).WithAttribute ("Include", name));
+			return itemGroup;
+		}
+
+		void Build (string projectFile, string target = "Build")
+		{
+			var binLog = Path.Combine (tempDirectory, "msbuild.binlog");
+			var psi = new ProcessStartInfo {
+				FileName = "msbuild",
```

Comment:
Yep, looks like `msbuild` is not in the path on the build agent so I have to "find" MSBuild on Windows.